src/main/host/descriptor/tcp.c

```diff
@@ -1431,7 +1431,7 @@ gboolean tcp_isValidListener(TCP* tcp) {
  */
 gboolean tcp_isListeningAllowed(TCP* tcp) {
     MAGIC_ASSERT(tcp);
-    if(tcp->state == TCPCS_NONE && tcp->flags == TCPF_NONE) {
+    if ((tcp->state == TCPS_CLOSED || tcp->state == TCPS_LISTEN) && tcp->flags == TCPF_NONE) {
```

Comment:
In addition to adding `TCPS_LISTEN` to the set of allowed listening states, I also make the change of `TCPCS_NONE` -> `TCPS_CLOSED`.

`TCPCS_NONE` is a TCP child state, not a TCP state. I think the correct state to check for is `TCPS_CLOSED`, which just happens to alias to the same value as `TCPCS_NONE(=0)`.